zebra-chain/src/equihash_solution.rs

```diff
@@ -30,7 +30,7 @@ pub struct EquihashSolution(
 
 impl EquihashSolution {
     /// Validate an equihash solution
-    pub fn is_valid(&self, input: &[u8], nonce: &[u8]) -> bool {
+    pub(crate) fn is_valid(&self, input: &[u8], nonce: &[u8]) -> bool {
```

Comment:
Suggestion: Please specify explicit sizes or types here. If we don't have a specific field or type, let's make the size into a constant.

```suggestion
    const INPUT_LENGTH: usize = 108;
    pub(crate) fn is_valid(&self, input: [u8; INPUT_LENGTH], nonce: [u8; 32]) -> bool {
```

Reply:
I don't think it's worth trying to make `input` an array over a slice, unless we never work with slices of data we will always have to do a conversion from a slice to an array which can only be done through `try_from`, which is already a runtime check, on top of that `try_from` is only implemented for arrays up to 32 elements in size, once we get const generics this will be better but for now I think its preferable to just have an assertion to validate the `INPUT_LENGTH` because either way this code only gets called in one place where we control the construction of `input`, and in all other cases its just test code and we'd have to do the runtime conversion anyways.

Reply:
Also the subsequent request to just do the tests with the `is_equihash_solution_valid` inspired me to just remove this function entirely so I think this is a moot point now